components/channel_notifications_modal/components/desktop_notification_section.test.jsx

```diff
@@ -0,0 +1,124 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import {shallow} from 'enzyme';
+
+import {NotificationLevels} from 'utils/constants';
+
+import DesktopNotificationSection from 'components/channel_notifications_modal/components/desktop_notification_section.jsx';
+import * as Utils from '../../../utils/utils';
+
+describe('components/channel_notifications_modal/DesktopNotificationSection', () => {
+    const baseProps = {
+        onChange: () => {}, //eslint-disable-line no-empty-function
+        onSubmit: () => {}, //eslint-disable-line no-empty-function
+        serverError: '',
+        onCollapseSection: () => {}, //eslint-disable-line no-empty-function
+        memberNotifyLevel: NotificationLevels.DEFAULT,
+        globalNotifyLevel: NotificationLevels.DEFAULT,
+        memberDesktopSound: undefined,
+        globalDesktopSound: 'true',
+        memberDesktopNotificationSound: undefined,
+        globalDesktopNotificationSound: 'Bing',
+        handleUpdateDesktopSound: () => {}, //eslint-disable-line no-empty-function
+        handleUpdateDesktopNotificationSound: () => {}, //eslint-disable-line no-empty-function
+    };
+
+    beforeEach(() => {
+        jest.clearAllMocks();
+    });
+
+    test('should match snapshot default global settings', () => {
+        const wrapper = shallow(
+            <DesktopNotificationSection {...baseProps}/>,
+        );
+
+        expect(wrapper).toMatchSnapshot();
+    });
+
+    test('should show notification options', () => {
+        const userAgentMock = jest.requireMock('../../../utils/user_agent');
+        userAgentMock.isDesktopApp.mockImplementationOnce(() => false);
+
+        jest.spyOn(Utils, 'hasSoundOptions').mockImplementationOnce(() => true);
```

Comment:
Instead of adding a mock here, it might be a bit easier to `jest.mock` these modules at the top level and then import them which lets you directly change the mocks without spying on Utils in every test.

```javascript
import * as UserAgent from 'utils/user_agent';
import * as Utils from 'utils/utils';

jest.mock('utils/user_agent'); // Mocks everything
jest.mock('utils/utils', () => ({
    ...jest.requireActual('utils/utils'),
    hasSoundOptions: jest.fn(() => true),
});

test('without sound options', () => {
    Utils.hasSoundOptions.mockReturnValue(false);
    // TODO test me
});
```

Reply:
Also, you should be able to simplify these import statements by leaving off the `.jsx` extension and by using absolute imports. Not strictly necessary, but it saves some typing and will make it easier when the extensions change as we migrate more code to TypeScript in the future